### ISSUE TITLE
cmake error

### ISSUE THREAD

**Issue by beria-moon** [TadasBaltrusaitis/OpenFace#1102]
/usr/bin/ld: /lib/x86_64-linux-gnu/libp11-kit.so.0: undefined reference to `ffi_closure_alloc@LIBFFI_CLOSURE_7.0'
/usr/bin/ld: /lib/x86_64-linux-gnu/libp11-kit.so.0: undefined reference to `ffi_type_uint8@LIBFFI_BASE_7.0'
/usr/bin/ld: /lib/x86_64-linux-gnu/libp11-kit.so.0: undefined reference to `ffi_prep_closure_loc@LIBFFI_CLOSURE_7.0'
/usr/bin/ld: /lib/x86_64-linux-gnu/libp11-kit.so.0: undefined reference to `ffi_type_uint64@LIBFFI_BASE_7.0'
/usr/bin/ld: /lib/x86_64-linux-gnu/libp11-kit.so.0: undefined reference to `ffi_type_pointer@LIBFFI_BASE_7.0'
/usr/bin/ld: /lib/x86_64-linux-gnu/libp11-kit.so.0: undefined reference to `ffi_prep_cif@LIBFFI_BASE_7.0'
collect2: error: ld returned 1 exit status
make[2]: *** [exe/FaceLandmarkImg/CMakeFiles/FaceLandmarkImg.dir/build.make:123: bin/FaceLandmarkImg] Error 1
make[1]: *** [CMakeFiles/Makefile2:331: exe/FaceLandmarkImg/CMakeFiles/FaceLandmarkImg.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
how to solve?

**Comment by brmarkus** [comment]
Are you within a virtual environment (with guaranteed package versions installed)? Or are you trying to build it natively? It looks like the version mismatch of `libffi`.

**Comment by staru09** [comment]
I am uanble to compile the library in Ubuntu 24.04, it says ``` Package 'g++-8' has no installation candidate```.
Is there a quick fix to this apart from running it in docker?

**Comment by kendrew06** [comment]
您的邮件我已收到，有需要会尽快回复This is an automatic reply, confirming that your e-mail was received.Thank you

**Comment by brmarkus** [comment]
> g++-8

Assuming you are following the Wiki from "https://github.com/TadasBaltrusaitis/OpenFace/wiki/Unix-Installation#dependency-installation"...
You could try to use newer versions of those packages which can't be installed on something new like Ubuntu 24.04. However, there will be many libraries in the meantime... OpenFace has many, many dependencies.

The MASTER branch is quite old and doesn't seem to still be maintained...

You might want to have a look into alternatives, like under "https://github.com/cmusatyalab/openface"

**Comment by staru09** [comment]
> You might want to have a look into alternatives, like under "https://github.com/cmusatyalab/openface"

Basically I am trying to run this specific [code](https://github.com/zeroQiaoba/MERTools/blob/master/MER2024/feature_extraction/visual/extract_openface.py) for a research project. The main goal is to extract key frame from a video based on maximum intensity  of action units. can you suggest some alternate to openface library.

**Comment by brmarkus** [comment]
No, can't suggest alternatives.
However, it runs great in a Docker container environment.

**Comment by staru09** [comment]
Yes I tried running it with docker, it worked fine. Will use that only.

Thanks for this amazing library though.

**Comment by brmarkus** [comment]
Don't thank me, I'm a user, too.
